txjuju/cli.py

```diff
@@ -17,6 +17,31 @@
 from .errors import CLIError
 
 
+def _find_best_juju(supported):
+    for juju in supported:
+        try:
+            _utils.Executable.find(juju)
+        except _utils.ExecutableNotFoundError:
+            continue
+        else:
+            return juju
+    else:
+        return supported[0]
+
+
+JUJU1 = _find_best_juju([
+        "juju-1",
+        "juju-1.25",
+        ])
+JUJU2 = _find_best_juju([
+        "juju-2",
+        "juju-2.0",  # the latest production release
+        # (lp:1650644) For now we must accommodate varying binary paths.
```

Comment:
Doing it at module-level is the point.  It's a one-time cost to populate a module-level constant with several possible values.  The alternative is to do the lookup in Juju2CLI.__init__.  However, in that case the lookup cost would be applied to every place that creates a new Juju2CLI, even though the value will be the same every time.  This approach isn't considered poor form.